include/ghex/transport_layer/mpi/hwtopo.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+#ifndef INCLUDED_GHEX_TL_MPI_HWTOPO_HPP
+#define INCLUDED_GHEX_TL_MPI_HWTOPO_HPP
+
+#include <vector>
+#include <unordered_set>
+#include "./error.hpp"
+
+namespace gridtools {
+    namespace ghex {
+        namespace tl {
+            namespace mpi {
+                
+                /** @brief Class representing node (shared memory) topology. */
+                class hwtopo {
```

Comment:
If we'll use hwloc this name could become confusing. Should we switch to another name line "rank_topology" ?

Reply:
yes, will do